packages/react-ui-core/src/DatePicker/Calendar.js

```diff
@@ -43,6 +48,34 @@ export default class Calendar extends PureComponent {
     }
   }
 
+  get daysOfWeek() {
+    const daysOfWeek = []
+    const monthStart = startOfMonth(this.state.date)
+    const dow = getDay(startOfMonth(this.state.date))
+    const firstDate = subDays(monthStart, dow)
+
+    for (let i = 0; i <= 6; i += 1) {
+      daysOfWeek.push(<div key={`dow-${i}`}>{format(addDays(firstDate, i), 'EEEEEE')}</div>)
+    }
+
+    return daysOfWeek
+  }
+
+  get daysOfMonth() {
+    const days = []
+
+    // NOTE: consitently create 5 rows so the calendar pads shorter months
+    for (let i = 0; i <= 5; i += 1) {
+      days.push(
+        <div key={`date-row-${i}`}>
```

Comment:
Since the normal dates of the month are all `button` elements (an inline element) this should probably also be an inline element (`span` instead of a `div`). For example, look at the HTML without styling, you don't want the days to appear one per line, you want them to appear all on the same line like the other dates.

Reply:
I believe this should be a block level element as it contains an entire week, not just a day.

Reply:
I think you're right on this one @triedal. I would change the days of the week though. I'll try to name some of this stuff better in the ui-ux version and it's somewhat confusing lol.